### PR TITLE
Infra proxy API to get the organisation of chef server

### PR DESCRIPTION
**Pull request by sonali523**
### :nut_and_bolt: Description: What code changed, and why?
List the organisation from the chef server
To use the web ui key in request to the chef server, I have added one parameter ```isWebuiKey```.
Based on this parameter we are adding one header as ```X-Ops-Request-Source:web``` in the go-chef library while making request to the chef server.
For go-chef changes, please refer this PR https://github.com/chef/go-chef/pull/30

### :chains: Related Resources
https://github.com/chef/automate/issues/6161
https://github.com/chef/go-chef/issues/29

### :+1: Definition of Done
I have added an API to list the organisation from the chef server
and 
### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER IDs
3. Add the web ui key in the request body

Request
This is a post request.
```curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/automateinfraserverorgs'```

request body
```
{
    "webui_key":"WEBUI_KEY"
}
```

Response

```
{
    "orgs": [
        {
            "id": "test",
            "name": "test",
            "admin_user": "",
            "credential_id": "",
            "server_id": "local-dev",
            "projects": []
        }
    ]
}
```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/infra-proxy-service/server/orgs.go [hunk 1]
```diff
@@ -210,6 +210,44 @@ func (s *Server) ResetOrgAdminKey(ctx context.Context, req *request.ResetOrgAdmi
...
+	}
+	orgsList, err := c.client.Organizations.List()
+	if err != nil {
+		return nil, ParseAPIError(err)
+	}
+
+	// Save organisations in backend DB
+	orgs := []storage.Org{}
+	for key := range orgsList {
+		org, err := s.service.Storage.StoreOrg(ctx, key, key, "pivotal", "", req.ServerId, nil)
```
Comment (kalroy) [inline components/infra-proxy-service/server/orgs.go:239]: I guess we can keep the org admin user name as blank which is default

File: components/infra-proxy-service/server/proxy.go [hunk 2]
```diff
@@ -160,6 +162,20 @@ func GetOrgAdminKeyFrom(secret *secrets.Secret) string {
 	return adminKey
 }
 
+// GetWebuiKeyFrom returns Webuikey
+func GetWebuiKeyFrom(secret *secrets.Secret) string {
```
Comment (kalroy) [inline components/infra-proxy-service/server/proxy.go:166]: This work is duplicate from #6364

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/infra-proxy-service/server/orgs.go (hunk 1 above)
Comment (sonali523) [reply components/infra-proxy-service/server/orgs.go]: Yup, did the changes.

File: components/infra-proxy-service/server/proxy.go (hunk 2 above)
Comment (sonali523) [reply components/infra-proxy-service/server/proxy.go]: Make the changes to use a single function for server and organisation key from secrets as `GetAdminKeyFrom(secret)`. After merging #6364 will keep only single function from both PRs.

**Comment by kalroy** [comment]
@sonali523 Please have a look at the pipeline and it is failing with repo-health status

**Comment by sonali523** [comment]
> @sonali523 Please have a look at the pipeline and it is failing with repo-health status

@kalroy checking it

**Comment by sonali523** [comment]
> > @sonali523 Please have a look at the pipeline and it is failing with repo-health status
> 
> @kalroy Checking it

@kalroy repo health is up now. The pipeline is failing for infra-proxy-service in server test cases which we are going to cover in server related APIs.

**Review by vinay033 — APPROVED** [review]

**Review by kalroy — APPROVED** [review]

**Pull request merged**